react/Product.js

```diff
@@ -0,0 +1,138 @@
+import React, { memo } from 'react'
+import { useRuntime } from 'vtex.render-runtime'
+import PropTypes from 'prop-types'
+import { pathOr, path, sort, last, flatten } from 'ramda'
+
+const getPrice = path(['commertialOffer', 'Price'])
+const getAvailableQuantity = pathOr(0, ['commertialOffer', 'AvailableQuantity'])
+
+const sortByPriceAsc = sort((itemA, itemB) => getPrice(itemA) - getPrice(itemB))
+
+const isSkuAvailable = sku => getAvailableQuantity(sku) > 0
+
+const lowHighForSellers = sellers => {
+  const sortedByPrice = sortByPriceAsc(sellers)
+  const withStock = sortedByPrice.filter(isSkuAvailable)
+  if (withStock.length === 0) {
+    return {
+      low: sortedByPrice[0],
+      high: last(sortedByPrice),
+    }
+  }
+
+  return {
+    low: withStock[0],
+    high: last(withStock),
+  }
+}
+
+const getSKUAvailabilityString = seller =>
+  isSkuAvailable(seller)
+    ? 'http://schema.org/InStock'
+    : 'http://schema.org/OutOfStock'
+
+const parseSKUToOffer = (item, currency) => {
+  const { low: seller } = lowHighForSellers(item.sellers)
+
+  const availability = getSKUAvailabilityString(seller)
+  const price = getPrice(seller)
+
+  // When a product is not available the API can't define its price and returns zero.
+  // If we set structured data product price as zero, Google will show that the
+  // product it's free (wrong info), but out of stock.
+  // It's better just not return any offer in that case.
+  if (availability === 'http://schema.org/OutOfStock' && price === 0) {
+    return null
+  }
+
+  const offer = {
+    '@type': 'Offer',
+    price,
+    priceCurrency: currency,
+    availability: getSKUAvailabilityString(seller),
+    sku: item.itemId,
+    itemCondition: 'http://schema.org/NewCondition',
+    priceValidUntil: path(['commertialOffer', 'PriceValidUntil'], seller),
+    seller: {
+      '@type': 'Organization',
+      name: seller.sellerName,
+    },
+  }
+
+  return offer
+}
+
+const getAllSellers = items => {
+  const allSellers = items.map(i => i.sellers)
```

Comment:
I would keep `i` for indexes rather than the element. Maybe use `el` or `item` or `element` like below?